src/unix/netbsd.c

```diff
@@ -64,34 +64,24 @@ void uv_loadavg(double avg[3]) {
 
 
 int uv_exepath(char* buffer, size_t* size) {
-  /* Intermediate buffer, retrieving partial path name does not work
-   * As of NetBSD-8(beta), vnode->path translator does not handle files
-   * with longer names than 31 characters.
-   */
-  char int_buf[PATH_MAX];
-  size_t int_size;
   int mib[4];
 
   if (buffer == NULL || size == NULL || *size == 0)
     return UV_EINVAL;
 
+  errno = 0;
   mib[0] = CTL_KERN;
   mib[1] = KERN_PROC_ARGS;
   mib[2] = -1;
   mib[3] = KERN_PROC_PATHNAME;
-  int_size = ARRAY_SIZE(int_buf);
 
-  if (sysctl(mib, 4, int_buf, &int_size, NULL, 0))
+  if (sysctl(mib, ARRAY_SIZE(mib), buffer, size, NULL, 0) != 0 && errno != ENOSPC)
     return UV__ERR(errno);
 
-  /* Copy string from the intermediate buffer to outer one with appropriate
-   * length.
-   */
-  /* TODO(bnoordhuis) Check uv__strscpy() return value. */
-  uv__strscpy(buffer, int_buf, *size);
-
-  /* Set new size. */
-  *size = strlen(buffer);
+  /* In case the buffer might not be C-string complete, adding trailing end */
+  if (!errno)
```

Comment:
Then I have a style suggestion:
```suggestion
  if (errno != 0)
```
And maybe move the `errno = 0;` to just before the `sysctl()`.

Reply:
Hmm not same semantic !errno && errno != 0. the former means decrease only when it s happy path. Agreed with the rest.

Reply:
While there, is there a rule to preserve errno from libuv calls? Another option would be to save original errno, set temporary errno to 0 and restore the original one before exit.

Something like:

```
int save_errno = errno;
errno = 0;
sysctl(...)
errno =  save_errno;
return 0;
```

In real life it might not matter, but it could save some debugging in bad style coding.

Reply:
As proof the unit tests do not have same outcome in the two cases.

Reply:
> Hmm not same semantic

Sorry yeah, that was a typo and shoulde have been `errno == 0`.